src/agentscope/rag/_reader/_word_reader.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+"""The Word reader to read and chunk Word documents."""
+import hashlib
+from typing import Literal
+
+from ._reader_base import ReaderBase
+from ._text_reader import TextReader
+from .._document import Document
+
+
+class WordReader(ReaderBase):
+    """The Word reader that splits text into chunks by a fixed chunk size."""
+
+    def __init__(
+        self,
+        chunk_size: int = 512,
+        split_by: Literal["char", "sentence", "paragraph"] = "sentence",
+    ) -> None:
+        """Initialize the Word reader.
+
+        Args:
+            chunk_size (`int`, default to 512):
+                The size of each chunk, in number of characters.
+            split_by (`Literal["char", "sentence", "paragraph"]`, default to \
+            "sentence"):
+                The unit to split the text, can be "char", "sentence", or
+                "paragraph". The "sentence" option is implemented using the
+                "nltk" library, which only supports English text.
+        """
+        if chunk_size <= 0:
+            raise ValueError(
+                f"The chunk_size must be positive, got {chunk_size}",
+            )
+
+        if split_by not in ["char", "sentence", "paragraph"]:
+            raise ValueError(
+                "The split_by must be one of 'char', 'sentence' or "
+                f"'paragraph', got {split_by}",
+            )
+
+        self.chunk_size = chunk_size
+        self.split_by = split_by
+
+        # To avoid code duplication, we use TextReader to do the chunking.
+        self._text_reader = TextReader(
+            self.chunk_size,
+            self.split_by,
+        )
+
+    async def __call__(
+        self,
+        word_path: str,
+    ) -> list[Document]:
+        """Read a Word document, split it into chunks, and return a list of
+        Document objects.
+
+        Args:
+            word_path (`str`):
+                The input Word document file path (.docx file).
+        """
+        try:
+            from docx import Document as DocxDocument
+        except ImportError as e:
+            raise ImportError(
+                "Please install python-docx to use the Word reader. "
+                "You can install it by `pip install python-docx`.",
+            ) from e
+
+        # Load the Word document
+        doc = DocxDocument(word_path)
+
+        # Extract text from all paragraphs
+        gather_texts = []
+        for paragraph in doc.paragraphs:
+            text = paragraph.text.strip()
+            if text:  # Only add non-empty paragraphs
+                gather_texts.append(text)
+
+        # Extract text from tables if any
+        for table in doc.tables:
```

Comment:
We should consider to reserve the order of the text and table content, rather than placing the table behind the text content